### ISSUE TITLE
Reusing documentation snippets

### ISSUE THREAD

**Issue by eseiler** [doxygen/doxygen#8390]
I am wondering if there is a way to reuse recurring documentation, e.g. in the form of doxygen snippets.

**Describe the bug**
For example, we have some documentation that looks like:

```cpp
/*!\brief A fine struct.
 *
 * \details
 *
 * Some descriptions.
 *
 * | Column | Column | Column |
 * |--------|--------|--------|
 * |  Cell  |  Cell  |  Cell  |
 * |  Cell  |  Cell  |  Cell  |
 * |  Cell  |  Cell  |  Cell  |
 */
struct fine_struct{};

/*!\brief Another struct.
 *
 * \details
 *
 * Some other description that differs from fine_struct, but uses the same table.
 *
 * | Column | Column | Column |
 * |--------|--------|--------|
 * |  Cell  |  Cell  |  Cell  |
 * |  Cell  |  Cell  |  Cell  |
 * |  Cell  |  Cell  |  Cell  |
 */
struct another_struct{};
```

Of course in different files, with different detailed description, but they share some, but not all, elements (e.g. only the table is common).

**Expected behavior**
We would like to be able to do something like this:

```cpp
/*!\brief Yet another struct.
 *
 * \details
 *
 * Yet another description.
 *
 * \include{doc} /dev/shm/tables/table.md
 */
struct yet_another_struct{};

/*!\brief One last struct.
 *
 * \details
 *
 * One last description that differs from yet_another_struct, but uses the same table.
 *
 * \include{doc} /dev/shm/tables/table.md
 */
struct one_last_struct{};
```

But this currently does not work:

**Screenshots**
![doxy report](https://user-images.githubusercontent.com/12967715/108201971-c270da00-7120-11eb-9b2c-47ffa4c95c1a.PNG)

**To Reproduce**
[example.tar.gz](https://github.com/doxygen/doxygen/files/5995485/example.tar.gz)
The `\include{doc} /dev/shm/tables/table.md` (`include.hpp`) is currently hardcoded and may need to be changed.

**Version**
`1.9.1 (ef9b20ac7f8a8621fcfc299f8bd0b80422390f4b)`

**Comment by albert-github** [comment]
It looks like the included markdown commands are not going through the markdown processor and thus are literally included into the code. This is one of the (non mentioned) exceptions in the Note with the `\include{doc}` command.

You could use instead of the markdown code the corresponding HTML code.
In this case:
```
<table class="markdownTable">
  <tr class="markdownTableHead">    <th class="markdownTableHeadNone"> Column \ilinebr </th>    <th class="markdownTableHeadNone"> Column \ilinebr </th>    <th class="markdownTableHeadNone"> Column \ilinebr </th>  </tr>
<tr class="markdownTableRowOdd">    <td class="markdownTableBodyNone"> Cell \ilinebr </td>    <td class="markdownTableBodyNone"> Cell \ilinebr </td>    <td class="markdownTableBodyNone"> Cell \ilinebr </td>  </tr>
<tr class="markdownTableRowEven">    <td class="markdownTableBodyNone"> Cell \ilinebr </td>    <td class="markdownTableBodyNone"> Cell \ilinebr </td>    <td class="markdownTableBodyNone"> Cell \ilinebr </td>  </tr>
<tr class="markdownTableRowOdd">    <td class="markdownTableBodyNone"> Cell \ilinebr </td>    <td class="markdownTableBodyNone"> Cell \ilinebr </td>    <td class="markdownTableBodyNone"> Cell \ilinebr </td>  </tr></table>
```

One can obtain this converted table by feeding just a markdown file to `doxygen -d markdown` catch the output and create a new file out of this. Note that the `\ilinebr` can be removed and are only present in the original for doxygen specific internal reasons (correct line numbers with warnings). Also note that the direct output of the `doxygen -d markdown` cannot be used as there might be some other "decoration" around (e.g page or section commands etc.).

It would be good to investigate whether it would be possible to feed the code of `\include{doc}` through the doxygen markdown processor or that there would be other complications..

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8395

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
